cpp/include/logistic_regression.h

```diff
@@ -38,28 +38,32 @@ namespace Nice {
 // Abstract class of common logistic regression functions
 template<typename T>
 class LogisticRegression {
+ private:
+  static Matrix<T> Sigmoid(const Matrix<T> &x){
```

Comment:
The sigmoid function takes in an entire matrix of values. I could do it element wise but I wanted to reduce the number of for-loops such that as much of the program as possible could run in parallel through the Eigen library. I can revert the code to the original for-loop if you would like.